app/models/shipment.rb

```diff
@@ -1,4 +1,10 @@
 class Shipment < ApplicationRecord
   belongs_to :company
-  belongs_to :shipment_item
+  has_many :shipment_items
+
+  def group_shipment_items
+    shipment_items.group(:description).order('count_all desc').count.map do |item|
+      {description: item[0], count: item[1]}
```

Comment:
heyhey ~ how to make it more readable ?

Reply:
can I refactor like this?
```ruby
def group_shipment_items
  shipment_items.group(:description).order('count_all desc').count
end

def set_shipment_items
  group_shipment_items.map do |item|
    {description: item[0], count: item[1]}
  end
end
```

Reply:
hey hey ~ nope~
`item[0]` and `item[1]` is not readable~~~ you can try to find another way to show it ~~

hint try to change `|item|`

Reply:
```ruby
def set_shipment_items
  group_shipment_items.map do |item, count|
    {description: item, count: count}
  end
end
```

got it!
I should use `key` and `value`
cause the method `group_shipment_items` returns a `Hash`